queries/prolog/highlights.scm

```diff
@@ -0,0 +1,43 @@
+(comment) @comment @spell
+
+(atom) @constant
+
+((atom) @boolean
+  (#any-of? @boolean "true" "false"))
+
+(functional_notation
+  function: (atom) @function.call)
+
+(integer) @number
+
+(float_number) @number.float
+
+(directive_head) @operator
+
+(operator_notation
+  operator: (_) @operator)
```

Comment:
Thank you, I think this PR is almost good to go! One thing: 
```suggestion
  operator: _ @operator)
```
This way anonymous nodes can be captured. Also, would you mind updating the grammar (and then the lockfile) so that things like:
```js
        prec.left(
          1,
          seq(
            $._term,
            field(fields.operator, /(:\-)|(\-\->)/),
            $._term,
          ),
        ),
```
can be indexed? Sadly regexes become invisible in the grammar and thus not query-able. Something like the following would allow for operators that use regex rules to be highlighted:
```js
        prec.left(
          1,
          seq(
            $._term,
            field(fields.operator, alias(/(:\-)|(\-\->)/, $.arrow)),
            $._term,
          ),
        ),
```
(or you can of course replace `$.arrow` with something quoted if you prefer the node be anonymous, your call)

Reply:
Should have fixed in https://github.com/nvim-treesitter/nvim-treesitter/pull/6810/commits/d95626829a0cda0c2b002dceb0745d6516ca37db